src/ray/raylet/local_object_manager.cc

```diff
@@ -85,26 +85,67 @@ void LocalObjectManager::FlushFreeObjectsIfNeeded(int64_t now_ms) {
   }
 }
 
+int64_t LocalObjectManager::SpillObjectsOfSize(int64_t num_bytes_required) {
+  if (RayConfig::instance().object_spilling_config().empty() ||
+      !RayConfig::instance().automatic_object_spilling_enabled()) {
+    return num_bytes_required;
+  }
+
+  RAY_LOG(INFO) << "Choosing objects to spill of total size " << num_bytes_required;
+  int64_t num_bytes_to_spill = 0;
+  auto it = pinned_objects_.begin();
+  std::vector<ObjectID> objects_to_spill;
+  while (num_bytes_to_spill < num_bytes_required && it != pinned_objects_.end()) {
+    num_bytes_to_spill += it->second->GetSize();
+    objects_to_spill.push_back(it->first);
+    it++;
+  }
+  RAY_LOG(INFO) << "Spilling objects of total size " << num_bytes_to_spill;
```

Comment:
How about we raise this to ERROR level so it's visible in the driver logs?

Reply:
(we can come up with a better reporting mechanism later on)

Reply:
Consider also adding some basic timing statistics here (bytes spilled in time t).